src/main/scala/com/fortysevendeg/scalacheck/datetime/j8/GenJ8.scala

```diff
@@ -0,0 +1,26 @@
+package com.fortysevendeg.scalacheck.datetime.j8
+
+import collection.JavaConverters._
+
+import org.scalacheck.Gen
+import org.scalacheck.Arbitrary.arbitrary
+
+import java.time._
+import java.time.temporal.ChronoUnit.MILLIS
+
+object GenJ8 {
+
+  val genZonedDateTime: Gen[ZonedDateTime] = for {
+    year <- Gen.choose(-292278994, 292278994)
+    month <- Gen.choose(1, 12)
+    maxDaysInMonth = Month.of(month).length(Year.of(year).isLeap)
+    dayOfMonth <- Gen.choose(1, maxDaysInMonth)
+    hour <- Gen.choose(0, 23)
```

Comment:
The documentation says that `java.time` implements dates following the `ISO-8601` standard. From what i read in the [wikipedia](https://en.wikipedia.org/wiki/ISO_8601), while I look for another source, this standard includes the following corner cases: 
* The use of the hour-minute pair `24:00` to indicate the midnight at the end of a day; for instance, the use of _until June 15th, 24:00_ to indicate until the end of the day June 15th.
* The use of the value `60` of seconds, to indicate [leap seconds](https://en.wikipedia.org/wiki/Leap_second).

I don't know if the `java.time` package includes those corner cases; if it does, perhaps they should be covered, either here or elsewhere.